html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Drupal\ghi_plans\Plugin\ParagraphHandler;
+
+/**
+ * Base class for paragraph handlers.
+ *
+ * @ParagraphHandler(
+ *   id = "plan_tab_switcher",
+ *   label = @Translation("Plan tab switcher")
+ * )
+ */
+class PlanTabSwitcher extends PlanBaseClass {
+
+  /**
+   * {@inheritdoc}
+   */
+  const KEY = 'plan_tab_switcher';
+
+  /**
+   * {@inheritdoc}
+   */
+  const DEFAULT_CONFIG = [
+    'show_count' => TRUE,
+    'render_all_children' => 0,
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preprocess(array &$variables, array $element) {
+    parent::preprocess($variables, $element);
+    $config = $this->getConfig();
+
+    /** @var \Drupal\node\Entity\Node $plan */
+    $plan = $this->parentEntity;
+    if ($plan->getType() !== 'plan' && $plan->getType() !== 'plan_para' && $plan->getType() !== 'plan_entity') {
+      return;
+    }
+
+    // Get parent if needed.
+    if ($plan->getType() === 'plan_entity') {
+      /** @var \Drupal\node\Entity\Node $plan */
+      $plan = $plan->field_plan->first->entity;
+    }
+
+    $overview_link = $this->t('Overview');
+    if (!$plan->isPublished()) {
+      if ($plan->access('view')) {
+        $overview_link = $plan->toLink($overview_link);
+      }
+    }
+    else {
+      $overview_link = $plan->toLink($overview_link);
+    }
+
+    $tabs = [
+      [
+        'title' => $this->t('Overview'),
+        'url' => $plan->toUrl(),
+        'attributes' => [
+          'class' => ['active'],
+        ],
+      ],
+    ];
+
+    // Get all plan entity nodes with plan as parent.
+    $plan_entities = \Drupal::entityTypeManager()->getStorage('node')->loadByProperties([
+      'type' => 'plan_entity',
+      'field_plan_id' => $plan->id(),
+    ]);
+
+    if ($plan_entities) {
+      $suffix = '';
+      if ($config['show_count']) {
+        $suffix = ' <span class="counter">' . count($plan_entities) . '</span>';
+      }
+
+      $tabs[] = [
+        'title' => $this->t('Clusters') . $suffix,
```

Comment:
The name of the GVE section can differ from plan to plan, this needs to be retrieved from the GVE prototype. As those prototypes should be the same across all GVEs for a plan, it's ok to use the first one. https://github.com/UN-OCHA/ghi-site/blob/8c0ee84468788445e085a92a516f8a8754e1e098/html/modules/custom/hpc_api/src/Helpers/ApiEntityHelper.php#L280 should give you what's needed then.
Also, `plan entities` are not clusters. `governing_entities` are clusters.

Reply:
I assumed all content types are in use